[2025-08-06T16:19:12.848+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-08-06T16:19:12.937+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spark_transform_topcv.submit_spark_job manual__2025-08-06T16:19:05.425114+00:00 [queued]>
[2025-08-06T16:19:12.947+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spark_transform_topcv.submit_spark_job manual__2025-08-06T16:19:05.425114+00:00 [queued]>
[2025-08-06T16:19:12.948+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2025-08-06T16:19:12.961+0000] {taskinstance.py:2330} INFO - Executing <Task(SparkSubmitOperator): submit_spark_job> on 2025-08-06 16:19:05.425114+00:00
[2025-08-06T16:19:12.970+0000] {standard_task_runner.py:63} INFO - Started process 65 to run task
[2025-08-06T16:19:13.027+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'spark_transform_topcv', 'submit_spark_job', 'manual__2025-08-06T16:19:05.425114+00:00', '--job-id', '43', '--raw', '--subdir', 'DAGS_FOLDER/etl_jobs_dag.py', '--cfg-path', '/tmp/tmpvvqr670d']
[2025-08-06T16:19:13.054+0000] {standard_task_runner.py:91} INFO - Job 43: Subtask submit_spark_job
[2025-08-06T16:19:13.500+0000] {task_command.py:426} INFO - Running <TaskInstance: spark_transform_topcv.submit_spark_job manual__2025-08-06T16:19:05.425114+00:00 [running]> on host 2a704dbc71ba
[2025-08-06T16:19:13.910+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='spark_transform_topcv' AIRFLOW_CTX_TASK_ID='submit_spark_job' AIRFLOW_CTX_EXECUTION_DATE='2025-08-06T16:19:05.425114+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-08-06T16:19:05.425114+00:00'
[2025-08-06T16:19:13.914+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-08-06T16:19:14.051+0000] {base.py:84} INFO - Using connection ID 'spark_default' for task execution.
[2025-08-06T16:19:14.056+0000] {spark_submit.py:351} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --conf spark.hadoop.fs.s3a.endpoint=http://minio:9000 --conf spark.hadoop.fs.s3a.access.key=minioadmin --conf spark.hadoop.fs.s3a.secret.key=****** --conf spark.hadoop.fs.s3a.path.style.access=true --conf spark.hadoop.fs.s3a.impl=org.apache.hadoop.fs.s3a.S3AFileSystem --jars /opt/custom-jars/hadoop-aws-3.3.2.jar,/opt/custom-jars/aws-java-sdk-bundle-1.12.367.jar --name arrow-spark --verbose /opt/spark-jobs/gold_transform.py
[2025-08-06T16:19:14.097+0000] {spark_submit.py:521} INFO - /home/***/.local/bin/spark-submit: line 27: /opt/spark/bin/spark-class: No such file or directory
[2025-08-06T16:19:14.100+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-08-06T16:19:14.114+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 156, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 452, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --conf spark.hadoop.fs.s3a.endpoint=http://minio:9000 --conf spark.hadoop.fs.s3a.access.key=minioadmin --conf spark.hadoop.fs.s3a.secret.key=****** --conf spark.hadoop.fs.s3a.path.style.access=true --conf spark.hadoop.fs.s3a.impl=org.apache.hadoop.fs.s3a.S3AFileSystem --jars /opt/custom-jars/hadoop-aws-3.3.2.jar,/opt/custom-jars/aws-java-sdk-bundle-1.12.367.jar --name arrow-spark --verbose /opt/spark-jobs/gold_transform.py. Error code is: 127.
[2025-08-06T16:19:14.149+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=spark_transform_topcv, task_id=submit_spark_job, run_id=manual__2025-08-06T16:19:05.425114+00:00, execution_date=20250806T161905, start_date=20250806T161912, end_date=20250806T161914
[2025-08-06T16:19:14.175+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 43 for task submit_spark_job (Cannot execute: spark-submit --master spark://spark-master:7077 --conf spark.hadoop.fs.s3a.endpoint=http://minio:9000 --conf spark.hadoop.fs.s3a.access.key=minioadmin --conf spark.hadoop.fs.s3a.secret.key=****** --conf spark.hadoop.fs.s3a.path.style.access=true --conf spark.hadoop.fs.s3a.impl=org.apache.hadoop.fs.s3a.S3AFileSystem --jars /opt/custom-jars/hadoop-aws-3.3.2.jar,/opt/custom-jars/aws-java-sdk-bundle-1.12.367.jar --name arrow-spark --verbose /opt/spark-jobs/gold_transform.py. Error code is: 127.; 65)
[2025-08-06T16:19:14.189+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-08-06T16:19:14.202+0000] {taskinstance.py:3498} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-08-06T16:19:14.204+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
